tt_metal/tt_metal.cpp

```diff
@@ -190,6 +190,15 @@ std::map<chip_id_t, Device *> CreateDevices(
 }
 
 void CloseDevices(std::map<chip_id_t, Device *> devices) {
+    // Global Sync across all devices in the pool.
+    // We need to ensure that commands sent to each device have been completed
+    // before closing any device + modifying routing info.
+    // If this is not done, non-blocking CCLs followed by a close will hang, since
+    // the main thread will modify device state while the CCL is running on device.
+    for (const auto &[device_id, dev] : devices) {
+        dev->synchronize(); // Synchronize worker queue
```

Comment:
@aliuTT do you want to clean up the extra sync in another issue?

Reply:
I will move CloseDevices to device pool